lib/test_scheduler.pm

```diff
@@ -0,0 +1,75 @@
+# Copyright © 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package test_scheduler;
+
+use base Exporter;
+use Exporter;
+
+use strict;
+use warnings;
+
+use File::Basename;
+use testapi qw(get_var set_var);
+use main_common 'loadtest';
+use YAML::Tiny;
+
+# use lib '/usr/share/openqa/lib';
+# use OpenQA::Utils qw(asset_type_from_setting locate_asset);
+
+our @EXPORT = qw(load_yaml_schedule);
+
+sub set_variables {
+    my ($var, $value) = @_;
+    $value =~ s/%(.*?)%/get_var($1)/eg;
+    # my $type = asset_type_from_setting($var, $value);
+    # $value = locate_asset($type, $value) if $type;
+    set_var($var, $value);
+}
+
+sub schedule_modules {
```

Comment:
Actually I agreed with you when you've said that we should split parsing and `loadtest` calls. So what's about returning list of modules here and schedule them separately?

Reply:
Cool, so then we are in the same page that is better to debugging purpose to have separated the logic at the cost to run the loop one more time. It makes sense. Changed for variables as well.